src/test/kotlin/com/exactpro/th2/common/message/TestMessageFilterUtils.kt

```diff
@@ -84,6 +104,27 @@ class TestMessageFilterUtils {
         Assertions.assertEquals(rootMessageFilterJson, objectMapper.writeValueAsString(toTreeTable))
     }
 
+    @ParameterizedTest
+    @MethodSource("additionalMetadata")
+    fun `valid root message filter to readable body collection conversion`(additionalMetadata: Map<String, String>) {
+        val expected =
+            readableRootMessageFilterJson.replace(ADDITIONAL_METADATA_TAG, mapToJsonConverter(additionalMetadata))
+        val toTreeTable = RootMessageFilter.newBuilder().apply {
+            messageType = "MsgType"
+            messageFilter = createMessageFilter()
+            metadataFilterBuilder.apply {
+                putPropertyFilters("propA", simplePropertyFilter("valA", NOT_EQUAL, true))
+                putPropertyFilters("propB", simplePropertyFilter("valB"))
+            }
+            comparisonSettingsBuilder.apply {
+                addIgnoreFields("fieldA")
+                addIgnoreFields("fieldB")
+            }
+        }.build().toReadableBodyCollection(additionalMetadata)
+        Assertions.assertNotNull(toTreeTable)
+        Assertions.assertEquals(expected, objectMapper.writeValueAsString(toTreeTable))
```

Comment:
As far as I remember there is no guarantee that the fields will be serialized in the same order on every machine. So, probably we need to serialize everything into **JsonNode**s and compare them